test-case-creation-strategy.mdx

```diff
@@ -0,0 +1,189 @@
+---
+title: Test case creation
+description: "How to best create test cases for your project"
+icon: "lightbulb"
+---
+
+## General Principles
+
+- start with a prompt
+- use `dependencies`
+- help the agent out if it went wrong
+- use `run locally` and `Playwright Codegen`
+- work on multiple test cases in parallel
+
+## Start with a prompt
+
+There are multiple ways to start with a prompt.
+
+1. On sign-up choose as many test cases as possible from the catalogue (login, link checks, change language)
+
+   <Frame caption="Select test cases on sign-up, screenshot 02/2024">
+     <img
+       src="images/sign-up-test-case-selection.png"
+       alt="select test cases on sign-up"
+       width="300px"
+     />
+   </Frame>
+
+2. Always use "Discover test cases" from the overview page. It gives you a headstart.
+
+   <Frame caption="Discover test cases from overiew page, screenshot 02/2024">
+     <img
+       src="images/discover-test-cases-card.png"
+       alt="discover test cases from overview page"
+       width="300px"
+     />
+   </Frame>
+
+3. Use "create custom test cases" from the overview page.
+   <Frame caption="Start test case creation formt he overview page, screenshot 02/2024">
+     <img
+       src="images/create-custom-test-cases-card.png"
+       alt="create customer test cases from the overview page"
+       width="300px"
+     />
+   </Frame>
+
+The sign-up procedure also creates a test cases to close the cookie banner, if applicable while the login test case typcially opens the door for the high value user flows.
+
+<Tip>
+
+**Advantages of starting with a prompt**
+
+- All these techniques will give you a couple of test cases to get startet with
+- with almost no effort.
+- Test case generation runs in parallel and in the background.
+
+</Tip>
+
+## Use dependencies
+
+Keep test cases short and to the point. Ideally each test case represents a small tasks you want to achieve. One technique to make this happen are dependencies.
+Typically the first test case you need is "accept all cookies". Second one is "login with username and password", third one might be "open profile page".
+The second test case would have to take care of cookies. The third test case would have to take care of cookies and login.
+This is because every test case runs in isolation and therefore always needs to be the full flow.
+This is where dependencies - or test case chaining - gets really valuable.
+Make "accept all cookies" a dependency of "login with username and password" and make "login with username and password" a dependency of "open profile page".
+This way the test runner as well as the AI Agent are using pre-play to execute dependencies before starting their actual work.
```

Comment:
this is the first time we talk about pre-play, I would probably just not mention it here, otherwise it's just confusing